ndb/src/google/cloud/ndb/model.py

```diff
@@ -1190,6 +1190,157 @@ def _apply_to_values(self, entity, function):
 
         return value
 
+    def _get_value(self, entity):
+        """Get the value for this property from an entity.
+
+        For a repeated property this initializes the value to an empty
+        list if it is not set.
+
+        Args:
+            entity (Model): An entity to get a value from.
+
+        Returns:
+            Any: The user value stored for the current property.
+
+        Raises:
+            UnprojectedPropertyError: If the ``entity`` is the result of a
+                projection query and the current property is not one of the
+                projected properties.
+        """
+        if entity._projection:
+            if self._name not in entity._projection:
+                raise UnprojectedPropertyError(
+                    "Property {} is not in the projection".format(self._name)
+                )
+
+        return self._get_user_value(entity)
+
+    def _delete_value(self, entity):
+        """Delete the value for this property from an entity.
+
+        .. note::
+
+            If no value exists this is a no-op; deleted values will not be
+            serialized but requesting their value will return :data:`None` (or
+            an empty list in the case of a repeated property).
+
+        Args:
+            entity (Model): An entity to get a value from.
+        """
+        if self._name in entity._values:
+            del entity._values[self._name]
+
+    def _is_initialized(self, entity):
+        """Ask if the entity has a value for this property.
+
+        This returns :data:`False` if a value is stored but the stored value
+        is :data:`None`.
```

Comment:
I'd like to know more about the rationale for that. Intuitively, it seems like it should return True, if it's supposed to be a test for whether a value is stored and not a test of the truthiness of the value.

Reply:
Assuming we're stuck with this because of history, though, this all looks good to me.

Reply:
1. Yes stuck because history
1. I think the name `_is_initialized` isn't great, but the purpose of the method isn't so bad. It's essentially answering "is a value set (if a value was supposed to be set)?"